tt_metal/hw/inc/dataflow_api.h

```diff
@@ -619,6 +627,23 @@ std::uint64_t get_noc_addr_helper(std::uint32_t noc_xy, std::uint32_t addr) {
     return ((uint64_t)(noc_xy) << NOC_ADDR_COORD_SHIFT) | addr;
 }
 
+FORCE_INLINE
+std::uint32_t  get_noc_exclude_region(
+    std::uint32_t exclude_start_x,
+    std::uint32_t exclude_start_y,
+    std::uint32_t exclude_dir_x,
+    std::uint32_t exclude_dir_y,
+    uint8_t noc = noc_index) {
+        /*
+            Get an encoding which contians the definition of the exclusion area
+        */
+        return (EXCLUDE_ENABLED << EXCLUDE_ENABLED_OFFSET |
+            DYNAMIC_NOC_DIRECTOIN(noc, exclude_dir_y) << EXCLUDE_DIRECTION_Y_OFFSET |
```

Comment:
typo

Reply:
Direction is only relevant to L-shape transactions, so no existing kernel should be affected. Updating the API to handle the mcast begin/end swap will force us to update all kernels - this is a later effort

Reply:
yep realized that and deleted the comment, from initially reading the comment I thought otherwise but I should have checked the code first :)